### ISSUE TITLE
Infinite Looping in startLoop sapi5.py

### ISSUE THREAD

**Issue by MikeKadoshnikov** [nateshmbhat/pyttsx3#308]
Using this package in a Robotics project, we have a function for speaking below:
    def say(self, text):
        &emsp; self.face.set_robot_state("talking")
        &emsp; if (self.voice._inLoop):
        &emsp; &emsp;self.voice.endLoop()
        &emsp; self.voice.say(text)
        &emsp; self.voice.runAndWait()
        &emsp; self.face.set_robot_state("idle")
        
and the runAndWait() function ends up running into an infinite loop. Going through PyCharm debugging I have narrowed it down to sapi5.py where it is getting stuck in startLoop(self) and looking at the code it is not obvious how this while loop is supposed to stop. Haven't found any resources on circumventing this issue, as its unfortunate since we wanted to use this package for our assignments, we will have to look for an alternative.
![pyttsx3issue](https://github.com/nateshmbhat/pyttsx3/assets/35706715/9ea919df-da91-4ab1-b856-498be475d98a)

**Comment by MikeKadoshnikov** [comment]
Found a way to have it not infinite loop but it requires us to initialize pyttsx3 every time we want the robot to say something, although it works it seems somewhat resource intensive since we have to reinitialize it everytime

**Comment by willwade** [comment]
So Im trying to replicate this.. 

```python
import pyttsx3

def say(theengine, text): 
    if (theengine._inLoop):
        theengine.endLoop()
    theengine.say(text)
    theengine.runAndWait()

engine = pyttsx3.init()
say(engine, "Hello, World!")

```

Have I got that right? That is not looping for me (although I am working in PR #329 - so test with that if you can just incase its me)

**Comment by MikeKadoshnikov** [comment]
Unfortunately we do not have access to the exact system we tested on. Although this issue primarily showed up on a raspberry pi 4, I believe we narrowed the issue down to global initialization with threading, somehow with the threading we couldn't reach the global engine. The solution was to initialize it every time we needed speech, and that seemed to work alright for what we needed it to do.
![image](https://github.com/user-attachments/assets/a57553b5-3a16-4afc-82f5-955d17021e38)
